src/Microsoft.TestPlatform.CommunicationUtilities/DataCollectionRequestHandler.cs

```diff
@@ -410,7 +410,8 @@ private void HandleAfterTestRunEnd(Message message)
             _requestData.MetricsCollection.Add(TelemetryDataConstants.InvokedDataCollectors, string.Join(",", invokedDataCollectorsForMetrics.ToArray()));
         }
 
-        var afterTestRunEndResult = new AfterTestRunEndResult(attachmentsets, invokedDataCollectors, _requestData.MetricsCollection.Metrics);
+        var afterTestRunEndResult = new AfterTestRunEndResult(attachmentsets, invokedDataCollectors,
+            _requestData.MetricsCollection.Metrics, _dataCollectionManager.GetTelemetryEvents());
```

Comment:
This should merge the data into metrics if possible, and not add any public apis.

Reply:
Adding all events into metrics can be theoretically possible but it is not elegant IMO. Currently metrics contains metrics of the execution and is used by Test Explorer to send only 1 telemetry event. By some other components those metrics can be used in different way. To add all telemetry events into metrics we would need to use 1 string and treat value in dictionary as list of tuples:

SPECIAL_KEY -> [ (name, Dictionary<string, object>()),  (name, Dictionary<string, object>()),  (name, Dictionary<string, object>())]

IMO keeping it as separate List is better. I extended public API exactly same as "InvokedDataCollector" so I think this change is fine.